packages/relay/src/utils.ts

```diff
@@ -212,4 +213,23 @@ export class Utils {
     // No configuration, use default behavior
     return [...rpcParams, requestDetails];
   }
+
+  /**
+   * Creates a new JsonRpcError with the request ID attached to assist with tracing and debugging.
+   *
+   * @param {JsonRpcError} error - The original JsonRpcError instance to bind the request ID to
+   * @param {string} requestId - The unique identifier of the request that generated this error
+   * @returns A new JsonRpcError instance with identical properties but with the request ID
+   *          prefixed to the error message
+   */
+  public static newJsonRpcErrorWithRequestId(error: JsonRpcError, requestId: string): JsonRpcError {
```

Comment:
I think this would be better placed in `JsonRpcError` class itself. It's easier to understand and more cohesive (no unrelated stuff in the `Utils` class). And given it would be part of `JsonRpcError` class, then it could be renamed to `newWithRequestId` or `withRequestId`.